pkg/health/health_daemonset.go

```
@@ -23,62 +23,70 @@ func getDaemonSetHealth(obj *unstructured.Unstructured) (*HealthStatus, error) {
-	health := HealthUnknown
+	desired := daemon.Status.DesiredNumberScheduled
+	updated := daemon.Status.UpdatedNumberScheduled
+	available := daemon.Status.NumberAvailable
+	ready := daemon.Status.NumberReady
+	unavailable := daemon.Status.NumberUnavailable
+
+	fullyUpdated := updated == desired
+	fullyAvailable := desired == 0 || (available == desired && ready == desired)
 
-	if daemon.Status.NumberAvailable == daemon.Status.DesiredNumberScheduled {
+	health := HealthUnknown
```

golangci-lint (2.11.3)

[error] ineffectual assignment to health

(ineffassign)